server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java

```diff
@@ -62,12 +71,40 @@ protected Logger getLogger() {
 
     @Override
     public void configure(HierarchicalConfiguration config) throws ConfigurationException {
-        defaultDomain = config.getString("defaultDomain", "localhost");
```

Comment:
It's more readable to handle one functional feature (default domain configuration) in only one method setting the defaultDomain and creating it when needed.